src/dns/rfc1035.cc

```diff
@@ -265,6 +265,11 @@ rfc1035NameUnpack(const char *buf, size_t sz, unsigned int *off, unsigned short
                 RFC1035_UNPACK_DEBUG;
                 return 1;
             }
+            /* before copying compression offset value, ensure it is inside the buffer */
+            if ((*off) + sizeof(s) > sz) {
+                RFC1035_UNPACK_DEBUG;
+                return 1;
+            }
             memcpy(&s, buf + (*off), sizeof(s));
             s = ntohs(s);
             (*off) += sizeof(s);
```

Comment:
AFAICT, the `if` statement misleadingly described as a "sanity check" below this line (I cannot attach a change request there) should now be removed or converted into a (paranoid) assertion:

```C++
if ((*off) + sizeof(s) > sz) return; // OK: new check
(*off) += sizeof(s);
if ((*off) > sz) return; // XXX: old check that we are now doing earlier
```

I suggest removing that old check (i.e. not adding a paranoid assertion).

Reply:
I agree it is effectively a duplicate check, and can be removed